lightning/src/ln/channelmanager.rs

```diff
@@ -14744,11 +14744,48 @@ where
 							debug_assert!(false, "Non-event-generating channel freeing should not appear in our queue");
 						}
 					}
+					let peer_state = peer_state.lock().unwrap();
+					let channel_id_matches = |updates: &(_, Vec<ChannelMonitorUpdate>)| {
+						updates.1.iter().any(|update| update.channel_id == Some(*channel_id))
+					};
+					if !peer_state.in_flight_monitor_updates.values().any(channel_id_matches) {
+						// If there are no pending `ChannelMonitorUpdate`s for this channel but we
+						// have pending post-update actions, its possible that one was left over
+						// from pre-0.1 payment claims where MPP claims led to a channel blocked on
+						// itself and later `ChannelMonitorUpdate`s didn't get their post-update
+						// actions run.
+						// This should only have happened for `PaymentClaimed` post-update actions,
+						// which we check here.
+						for action in actions.iter() {
+							if let MonitorUpdateCompletionAction::PaymentClaimed { .. } = action {
+							} else {
+								let logger =
+									WithContext::from(&args.logger, Some(node_id), None, None);
+								log_error!(logger, "Got blocked actions {:?} without a per-peer-state for {}", actions, node_id);
+								return Err(DecodeError::InvalidValue);
```

Comment:
This is all new code that adds a new error condition. 

>Here we fix this issue by specifically checking for dangling
`MonitorUpdateCompletionAction::PaymentClaim` entries and dropping
them if there is no corresponding channel or peer state entry.

How are they dropped now, compared to how it happens in main?